src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -181,16 +182,35 @@ export class NetworkAreaDiagramViewer {
         );
         return node?.svgId || null;
     }
-
+    private getTextNodeIdFromEquipmentId(equipmentId: string) {
+        const node: TextNodeMetadata | undefined = this.diagramMetadata?.textNodes.find(
+            (node) => node.equipmentId == equipmentId
+        );
+        return node?.svgId || null;
+    }
     public moveNodeToCoordinates(equipmentId: string, x: number, y: number) {
         const nodeId = this.getNodeIdFromEquipmentId(equipmentId);
         if (nodeId != null) {
             const elemToMove: SVGElement | null = this.container.querySelector('[id="' + nodeId + '"]');
             if (elemToMove) {
                 const newPosition = new Point(x, y);
                 this.onDragStart(elemToMove);
-                this.onDragEnd(newPosition, false);
+                this.onDragEnd(newPosition, false, false);
+            }
+        }
+    }
+    public moveTextNodeToCoordinates(equipmentId: string, x: number, y: number) {
+        const textnodeId = this.getTextNodeIdFromEquipmentId(equipmentId);
+        if (textnodeId != null) {
+            const elemToMove: SVGElement | null = this.container.querySelector('[id="' + textnodeId + '"]');
+            if (elemToMove) {
+                const newPosition = new Point(x, y);
+                this.onDragStart(elemToMove);
+                this.onDragEnd(newPosition, false, false);
             }
+            //Update css rules to diplay labels
+            const firstChild: HTMLElement = <HTMLElement>this.svgDraw?.node.firstChild;
+            this.injectDisplayCssRules(firstChild);
```

Comment:
What is the purpose of this ?

Reply:
the purpose is to reapply css rules to display labels